pages/dashboard/DisplayBookmarks.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import Link from 'next/link';
+import { useTranslation } from 'react-i18next';
+import { DeleteBinIcon } from '@remixicons/react/line';
+
+import { useMatrix } from '../../lib/Matrix';
+import { ServiceTable } from '../../components/UI/ServiceTable';
+import { useAuth } from '../../lib/Auth';
+import TextButton from '../../components/UI/TextButton';
+import LoadingSpinnerInline from '../../components/UI/LoadingSpinnerInline';
+import CopyToClipboard from '../../components/UI/CopyToClipboard';
+import Icon from '../../components/UI/Icon';
+
+/**
+ * DisplayBookmarks component for rendering and managing bookmarks within a specified space.
+ * This component displays a list of bookmarks, allowing users to remove them.
+ *
+
+ @param {string} bookmarkSpaceId - The ID of the space containing bookmarks.
+ @param {string} name - the name of the origin of the bookmark.
+ * @returns {JSX.Element|null} - A JSX element containing the list of bookmarks, or null if the space is not found.
+ */
+
+export default function DisplayBookmarks({ bookmarkSpaceId, name }) {
+    const auth = useAuth();
+    const matrix = useMatrix(auth.getAuthenticationProvider('matrix'));
+
+    const removeBookmark = async (parent, roomId) => {
+        await auth.getAuthenticationProvider('matrix').removeSpaceChild(parent, roomId);
+        await matrix.leaveRoom(roomId);
+    };
+
+    if (!matrix.spaces.get(bookmarkSpaceId)) return null;
+
+    return (
+        <>
+            { matrix.spaces.get(bookmarkSpaceId)?.children.map(roomId => {
+                return <Bookmark
+                    key={roomId}
+                    parent={bookmarkSpaceId}
+                    roomId={roomId}
+                    link={matrix.roomContents.get(roomId).body}
+                    name={matrix.rooms.get(roomId).name}
+                    origin={name}
+                    removeBookmark={removeBookmark}
+                />;
+            }) }
+        </>);
+}
+
+/**
+ * Bookmark component for rendering a single bookmark entry.
+ *
+ * @param {string} parent - The ID of the parent space.
+ * @param {string} roomId - The ID of the bookmarked room.
+ * @param {string} link - The link associated with the bookmark.
+ * @param {string} name - The name of the bookmarked room.
+ * @param {string} origin - The name of the origin of the bookmarked room.
+ * @param {Function} removeBookmark - A function to remove the bookmark.
+ * @returns {JSX.Element} - A JSX element representing a single bookmark entry.
+ *
+ * @TODO
+ * origin only works
+ */
+const Bookmark = ({ parent, roomId, link, name, origin, removeBookmark }) => {
+    const [removingBookmark, setRemovingBookmark] = useState(false);
+    const { t } = useTranslation();
+
+    const handleRemove = async (e) => {
+        e.preventDefault();
+        setRemovingBookmark(true);
+        await removeBookmark(parent, roomId)
```

Comment:
If we're passing the `parent` into the `<Bookmark` component, only to then call the `removeBookmark()` function that is defined in the actual "parent" component `<DisplayBookmarks` we can definitely save up on that.

Instead you can strip away both: `parent` and `removeBookmark` from this `<Bookmark>` component.

And above, in line 45 we can do:
```jsx
    removeBookmark={() => removeBookmark(roomId)}
```

Line 29 can be
```jsx
    const removeBookmark = async (roomId) => {
        await auth.getAuthenticationProvider('matrix').removeSpaceChild(bookmarkSpaceId, roomId);
        return matrix.leaveRoom(roomId);
    };
```